### PR TITLE
[thetanuts] Update to validate_bid: ensure wallet has sufficient collateral

### PR DESCRIPTION
**Pull request by ezoia-com**
Also updates create_offer to return correct swapId, which is the concatenation of the swapIndex and vault epoch

### CODE REVIEW

File: thetanuts/thetanuts/config.py [hunk 1]
```diff
@@ -276,6 +283,16 @@ def validate_bid(
...
 
+        bidding_token = w3.eth.contract(
+            w3.toChecksumAddress(vault.functions.COLLAT().call()),
+            abi=get_abi("ERC20"),
+        )
+
+        # Check for sufficient assets in wallet
+        assetBalance = bidding_token.functions.balanceOf(signer_wallet).call()
+        if assetBalance < sell_amount:
+            return {'errors': True, "messages": ["insufficient bidding token in wallet"]}
```
Comment (mdantonio) [inline thetanuts/thetanuts/config.py:294]: `errors` is expected to be an `int`
([here the definition](https://github.com/tradeparadigm/sdks/blob/3d0d997427152f60036132d0875b8f5fe4d01925/sdk_commons/sdk_commons/config.py#L30-L32))

So should correspond to the length of the messages list, if any

I propose to change the last three lines as:

```
if assetBalance < sell_amount:
    return {'errors': 1, "messages": ["insufficient bidding token in wallet"]}
if not isValid:
    return {'errors': 1, "messages": ["???????? bid is invalid"]}
return {'errors': 0}
```

Comment (ezoia-com) [reply thetanuts/thetanuts/config.py:294]: ok! i rearranged it such that it more clearly identifies the 

        # check sufficient tokens first
        # check signature and parameters
        ....
        except Exception: # Catches EVM revert, when signature submitted is invalid
            return {'errors': 1, "messages": ["signature invalid"]}

        if isValid:
          return {'errors': 0}
        else:
          return {'errors': 1, "messages": ["signature valid, bid parameters invalid"]}

File: thetanuts/thetanuts_test.py [hunk 2]
```diff
@@ -161,29 +161,28 @@ def get_env(variable: str) -> str:
...
+        * Decimal(pricePerContract)
+        * COLLAT_DECIMALS
+        / PARADIGM_DECIMALS
+    ),
+    buy_amount=int(Decimal(offer["availableSize"]) * COLLAT_DECIMALS / PARADIGM_DECIMALS),
+    referrer="0x" + "0" * 40,
+    signature=signed_bid,
+)
+
+if bidValidation["errors"] is False:
```
Comment (mdantonio) [inline thetanuts/thetanuts_test.py:182]: This should be 
```
if bidValidation["errors"] == 0:
```

now

File: thetanuts/thetanuts_test.py [hunk 3]
```diff
@@ -161,29 +161,28 @@ def get_env(variable: str) -> str:
...
-            * COLLAT_DECIMALS
-            / PARADIGM_DECIMALS
-        ),
-        buy_amount=int(Decimal(offer["availableSize"]) * COLLAT_DECIMALS / PARADIGM_DECIMALS),
-        referrer="0x" + "0" * 40,
-        signature=signed_bid,
-    )["errors"]
-    is False
-):
+bidValidation = thetanuts.validate_bid(
```
Comment (mdantonio) [inline thetanuts/thetanuts_test.py:164]: related to the line above (163)
return is errors: 0, no longer False

File: thetanuts/thetanuts_test.py (hunk 2 above)
Comment (ezoia-com) [reply thetanuts/thetanuts_test.py]: ok! resolved these

**Review by mdantonio — APPROVED** [review]

**Pull request merged**